docs/tools/platformio.mdx

```diff
@@ -0,0 +1,124 @@
+---
+custom_edit_url: null
+description: The gate is a major tool of the Luos ecosystem.
+---
+
+import { customFields } from '/docusaurus.config.js';
+import Tooltip from '@site/src/components/Tooltip.js';
+import Image from '@site/src/components/Images.js';
+import IconExternalLink from "@theme/IconExternalLink";
+
+# PlatformIO
+
+## What is PlatformIO?
+
+PlatformIO is a user-friendly and extensible integrated development environment with a set of professional development instruments, compilers and debuggers. PlatformIO provides modern and powerful features to simplify the creation and delivery of embedded products.
+That is the reason why Luos uses PlatformIO's project structure for its examples, allowing users to use and share easily services and developments on a unified environment. Also, Luos engine and some tools like the [gate](/docs/tools/gate) or the [inspector](/docs/tools/inspector) are available in the PlatformIO's registry. This registry allows user to link directly a project and a library, like Luos engine, by including dependencies in the configuration project file called _platformio.ini_.
+Learn more about <a href="https://platformio.org" target="_blank">PlatformIO and how to install it in VSCode <IconExternalLink width="10" /></a>.
+
+## PlatformIO's project structure
+
+Every PlatformIO's project has a <a href="https://docs.platformio.org/en/latest/projectconf/section_platformio.html#directory-options" target="_blank">basic folder project structure<IconExternalLink width="10"/></a> and must have at the root of the project the file <a href="https://docs.platformio.org/en/latest/projectconf/section_platformio.html#generic-options" target="_blank">_platformio.ini_ <IconExternalLink width="10" /></a>.
+
+Checkout the Luos tutorial on <a href="https://www.luos.io/tutorials/pio" target="_blank">how to create a Luos project with PlatformIO <IconExternalLink width="10" /></a>.
+
+## Luos configuration for _platformio.ini_
+
+Some build flags definition may be used to configure Luos engine when it is added in your project. Those definitions will take care of the Luos engine library by using a Luos script called when building a project with Luos engine. Those definitions must be set in _platformio.ini_.
+For more information, see the script _source_filter_script.py_ in <a href="https://github.com/Luos-io/luos_engine" target="_blank" rel="external nofollow">Luos engine <IconExternalLink width="10" /></a>.
```

Comment:
You should add a link to the script the same way this is done on gate and pipe.

Reply:
don't understand what you means?